src/response.rs

```diff
@@ -181,6 +183,38 @@ pub fn parse_get_gj_comments_response(response: &str) -> Result<Vec<LevelComment
         .collect()
 }
 
+pub fn parse_get_gj_comment_history_response(response: &str) -> Result<Vec<CommentHistory>, ResponseError> {
+    if response == "-1" {
+        return Err(ResponseError::NotFound)
+    }
+
+    let mut sections = response.split('#');
+
+    // The format here is very weird. We have a '|' separated list of (comment, user) pairs, and said
+    // pair is separated by a ':'
+
+    section!(sections)
+        .split('|')
+        .map(|fragment| {
+            let mut parts = fragment.split(':');
+
+            if let (Some(raw_comment), Some(raw_user)) = (parts.next(), parts.next()) {
+                let mut comment = CommentHistory::from_robtop_str(raw_comment)?;
+
+                comment.user = if raw_user == "1~~9~~10~~11~~14~~15~~16~" {
+                    None
+                } else {
+                    Some(CommentUser::from_robtop_str(raw_user)?)
+                };
+
+                Ok(comment)
+            } else {
+                Err(ResponseError::UnexpectedFormat)
+            }
```

Comment:
iirc this is copy pasted from the level comment processes so maybe there's a way to unify them

Reply:
Already unified in another commit that I will make shortly.